### DISCUSSION TITLE
Orange Document Map

### DISCUSSION THREAD

**Discussion by LiseBo** [biolab/orange3#5876] [Q&A]
I've used the Document Map widget in Orange to identify all countries mentioned in the corpus. There appears to be a bug in the document map: On the map itself some countries are not colored (high lighted) to indicate the number of documents linked to a country. For example South Africa is not highlighted at all, thus it shows that zero mention is made within my corpus about South Africa. However if I click on the country one can clearly see in the output that there are a number of references to that country. Also Corpus Viewer then provides a list of the documents related to that country. Is there somehow a way to get the map to color all the countries referenced in the corpus? Or is this a bug?

**Comment by ajdapretnar** [comment]
Could you provide some screenshots perhaps?

**Reply by LiseBo** [reply]
For example: Here is the document map, without selecting any country. Please note that, as an example of the issue, South Africa and Ethiopia is not colored at all, thus indicating zero representation in the data. 
![image](https://user-images.githubusercontent.com/101125233/157106594-c748c54d-9012-4152-bf21-d362fc6160fb.png)

If I select those countries though it is clear that there are representation in the data. Please see screenshots below which shows South Africa with 15 documents represented and Ethiopia with 40. After selecting the country I am able to see these documents in the corpus viewer widget. So it appears just the actual map coloring to be an issue. 
South Africa:
![image](https://user-images.githubusercontent.com/101125233/157107350-e016e491-f17d-403a-a451-7a806d4a0f05.png)
Ethiopia: 
![image](https://user-images.githubusercontent.com/101125233/157107438-d49c1b73-e067-4e73-87f9-cdd5cca1fae2.png)

Also is there a way to extract the data per all countries or selection of countries, as the only way I can see now is to extract per country?

**Reply by ajdapretnar** [reply]
I see. Could you perhaps submit a minimal reproducible example, so that we can find the issue? That is the data and the workflow where the issue appears (I cannot reproduce it locally).

As for the selection of countries - no. Document Map works by findings English names of countries in the text. It cannot add a new column with country name, because there can be many countries mentioned in the same text. So please note - the documents can overlap. One can mention several countries.

**Reply by LiseBo** [reply]
Thank you. Here is the data: 
[Uploading AQ Articles for Orange 27.2.csv…](
I am not sure how to send the workflow, but here is a screenshot:
![image](https://user-images.githubusercontent.com/101125233/157405648-be15d112-f8a9-4c71-a087-fe4abfd1178e.png)
The issue is the same even if I connect the Document map widget directly to the corpus, thus the preprocessing text does not have an effect. In the document Map settings the "regional attribute" is set to analyse the "abstract" data.
